google-cloud-bigquerystorage/src/test/java/com/google/cloud/bigquery/storage/v1/it/ITBigQueryStorageTest.java

```diff
@@ -329,6 +607,130 @@ public void testRangeType() throws InterruptedException {
     assertEquals(1, rowCount);
   }
 
+  @Test
+  public void testRangeTypeWrite()
+      throws InterruptedException, IOException, DescriptorValidationException {
+    // Create table with Range fields.
+    String tableName = "test_range_type_write";
+    TableId tableId = TableId.of(DATASET, tableName);
+    bigquery.create(TableInfo.of(tableId, StandardTableDefinition.of(RANGE_SCHEMA)));
+
+    TableName parentTable = TableName.of(projectName, DATASET, tableName);
+    RetrySettings retrySettings =
+        RetrySettings.newBuilder()
+            .setInitialRetryDelay(Duration.ofMillis(500))
+            .setRetryDelayMultiplier(1.1)
+            .setMaxAttempts(5)
+            .setMaxRetryDelay(Duration.ofMinutes(1))
+            .build();
+    try (JsonStreamWriter writer =
+        JsonStreamWriter.newBuilder(parentTable.toString(), RANGE_TABLE_SCHEMA)
+            .setRetrySettings(retrySettings)
+            .build()) {
+
+      // Write 4 rows of data to the table with and without unbounded values.
+      JSONArray data = new JSONArray();
+      for (String name : RANGE_TEST_VALUES_DATES.keySet()) {
+        JSONObject row = new JSONObject();
+        row.put("name", name);
+
+        JSONObject dateColumn = new JSONObject();
+        Range date = RANGE_TEST_VALUES_DATES.get(name);
+        if ((!date.getStart().isNull()) && (date.getStart().getStringValue() != null)) {
+          dateColumn.put("start", date.getStart().getStringValue());
+        }
+        if ((!date.getEnd().isNull()) && (date.getEnd().getStringValue() != null)) {
+          dateColumn.put("end", date.getEnd().getStringValue());
+        }
+        row.put("date", dateColumn);
+
+        JSONObject datetimeColumn = new JSONObject();
+        Range datetime = RANGE_TEST_VALUES_DATETIME.get(name);
+        if ((!datetime.getStart().isNull()) && (datetime.getStart().getStringValue() != null)) {
+          datetimeColumn.put("start", datetime.getStart().getStringValue());
+        }
+        if ((!datetime.getEnd().isNull()) && (datetime.getEnd().getStringValue() != null)) {
+          datetimeColumn.put("end", datetime.getEnd().getStringValue());
+        }
+        row.put("datetime", datetimeColumn);
```

Comment:
Try to use "dateTime" as field name instead and it should be able to pass.

Reply:
Updated.